mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -1629,8 +1646,35 @@ void mlir::rock::expandFlatFunctionArguments(
       logicalVal = arg;
       continue;
     }
-    TopDownTMBuilder flattener(b, nameList, logicalShapedTy.getShape(), loc);
-    flattener.unmerge("raw", 0, nameList, logicalShapedTy.getShape());
+    SmallVector<uint32_t> upperDims(logicalShapedTy.getRank());
+    std::iota(upperDims.begin(), upperDims.end(), 0);
+    SmallVector<uint32_t> nonUnitUpperDim;
```

Comment:
Looks like we don't have unit-tests for `transformMapUtils`
IMO we should have one. We rely a lot on transformMapUtils.